File: FoosballGames/FoosballGameCommandHandler.cs
using FoosballGames.Contracts;
using FoosballGames.Infrastructure.Messaging;

namespace FoosballGames;

public class FoosballGameCommandHandler : ICommandHandler<AddPointForTeam>, ICommandHandler<CreateFoosballGame>
{
    private readonly IFoosballGamesRepository _foosballGamesRepository;

    public FoosballGameCommandHandler(IFoosballGamesRepository foosballGamesRepository)
    {
        _foosballGamesRepository = foosballGamesRepository;
    }

    public async Task HandleAsync(AddPointForTeam command)
    {
        var game = await _foosballGamesRepository.Get(command.GameId);
        var newGame = game.AddPointForTeam(command.Team);
        await _foosballGamesRepository.Update(newGame);
    }

    public async Task HandleAsync(CreateFoosballGame command)
    {
        var game = FoosballGame.Initialize(command);
        await _foosballGamesRepository.Add(game);
    }
}

File: FoosballGames/FoosballGameCommandHandler.fs
namespace FoosballGames

open FoosballGames
open FoosballGames.Contracts
open FoosballGames.Infrastructure.Messaging

type FoosballGameCommandHandler(repository: IFoosballGamesRepository) =
    interface ICommandHandler<CreateFoosballGame> with
        member this.HandleAsync(command) =
            let game = FoosballGame.Initialize command
            repository.Add game
            
    interface ICommandHandler<AddPointForTeam> with
        member this.HandleAsync(command) =
            task {
                let! game = repository.Get command.GameId
                let newGame = game.AddPointForTeam command.Team
                do! repository.Update newGame
            }
        